dbms/src/Encryption/RateLimiter.cpp

```diff
@@ -289,45 +289,25 @@ void WriteLimiter::updateMaxBytesPerSec(Int64 max_bytes_per_sec)
 }
 
 ReadLimiter::ReadLimiter(
-    std::function<Int64()> getIOStatistic_,
+    std::function<Int64()> get_read_bytes_,
     Int64 rate_limit_per_sec_,
     LimiterType type_,
-    Int64 get_io_stat_period_us,
     UInt64 refill_period_ms_)
     : WriteLimiter(rate_limit_per_sec_, type_, refill_period_ms_)
-    , getIOStatistic(std::move(getIOStatistic_))
-    , last_stat_bytes(getIOStatistic())
-    , last_stat_time(now())
+    , get_read_bytes(std::move(get_read_bytes_))
+    , last_stat_bytes(get_read_bytes())
     , log(Logger::get("ReadLimiter"))
-    , get_io_statistic_period_us(get_io_stat_period_us)
 {}
 
 Int64 ReadLimiter::getAvailableBalance()
 {
-    TimePoint us = now();
-    // Not call getIOStatisctics() every time for performance.
-    // If the clock back, elapsed_us could be negative.
-    Int64 elapsed_us = std::chrono::duration_cast<std::chrono::microseconds>(us - last_stat_time).count();
-    if (get_io_statistic_period_us != 0 && elapsed_us < get_io_statistic_period_us)
-    {
-        return available_balance;
-    }
-
-    return refreshAvailableBalance();
-}
-
-Int64 ReadLimiter::refreshAvailableBalance()
-{
-    TimePoint us = now();
-    Int64 bytes = getIOStatistic();
+    Int64 bytes = get_read_bytes();
     if (bytes < last_stat_bytes)
     {
         LOG_FMT_WARNING(
             log,
-            "last_stat {}:{} current_stat {}:{}",
-            last_stat_time.time_since_epoch().count(),
+            "last_stat: {} current_stat: {}",
             last_stat_bytes,
-            us.time_since_epoch().count(),
             bytes);
     }
     else
```

Comment:
There is very likely that `bytes == last_stat_bytes` since `bytes` is calculated periodically. In this scenario,  calculation and metircs are unnecessary.

Reply:
get